bigtable/bttest/grpc_error_injection.go

```diff
@@ -0,0 +1,315 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This file provides interceptors to hook into grpc streaming calls
+and inject latency and errors into cbtemulator for testing
+*/
+package bttest
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	validStreamMethodSuffixes = []string{"ReadRows", "MutateRows"}
+)
+
+/*
+   Creates interceptors inject latency or errors into cbtemulator
+*/
+type EmulatorInterceptorBuilder struct {
+	LatencyTargets       latencyTargets
+	GrpcErrorCodeTargets grpcErrorCodeTargets
```

Comment:
not a golang expert, but this feels weird: the variable name is public, but the type is private

Reply:
Goal here is to ensure that LatencyTargets and GrpcErrorCodeTargets can be used by `flags.Var` in `cbtemulator.go`, while keeping the underlying types private.

It feels a bit weird to me as well - open to suggestions

Reply:
Agreed that it doesn't seem super well-encapsulated as is. Maybe you could use a constructor function, or a setter method for the private field? Or if you take my suggestion about using JSON then you can just have a constructor that takes the filename and then the fields can be private.

Reply:
+1 to using a constructor function. This way clients (not just the emulator) have a clear means to instantiating an `EmulatorInterceptor`, as opposed to just relying on an invocation of the emulator from the command line.